src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,196 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<T>(value);
+
+        if (size == 0) {
+            head = newNode;
+            tail = newNode;
+            size++;
+            return;
+        }
+
+        Node<T> prevTail = tail;
+        newNode.prev = tail;
+        tail = newNode;
+        prevTail.next = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
+        if (size == 0) {
+            add(value);
+            return;
+        }
+        Node<T> newNode = new Node<>(value);
+        if (index == 0) {
+            head.prev = newNode;
+            newNode.next = head;
+            head = newNode;
+            size++;
+            return;
+        }
+        Node<T> currentNode = head;
+        for (int i = 0; i < index - 1; i++) {
+            if (currentNode == null) {
+                add(value);
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+        if (currentNode == null || currentNode.next == null) {
+            add(value);
+            return;
+        }
```

Comment:
let's create a separate method getNode(index) as we gonna use the same logic also in methods set / get / remove